pkg/apis/stable/v1alpha1/fleetautoscaler.go

```diff
@@ -113,10 +128,72 @@ type FleetAutoscalerStatus struct {
 	ScalingLimited bool `json:"scalingLimited"`
 }
 
+// FleetAutoscaleRequest defines the request to webhook autoscaler endpoint
+type FleetAutoscaleRequest struct {
+	// UID is an identifier for the individual request/response. It allows us to distinguish instances of requests which are
+	// otherwise identical (parallel requests, requests when earlier requests did not modify etc)
+	// The UID is meant to track the round trip (request/response) between the Autoscaler and the WebHook, not the user request.
+	// It is suitable for correlating log entries between the webhook and apiserver, for either auditing or debugging.
+	UID types.UID `json:"uid"`
+	// Name is the name of the Fleet being scaled
+	Name string `json:"name"`
+	// Namespace is the namespace associated with the request (if any).
+	Namespace string `json:"namespace"`
+	// The Fleet's status values
+	Status FleetStatus `json:"status"`
+}
+
+// FleetAutoscaleResponse defines the response of webhook autoscaler endpoint
+type FleetAutoscaleResponse struct {
+	// UID is an identifier for the individual request/response.
+	// This should be copied over from the corresponding FleetAutoscaleRequest.
+	UID types.UID `json:"uid"`
+	// The targeted replica count
+	// Set to nil if no scaling should occur to the Fleet
+	Replicas *int32 `json:"replicas"`
```

Comment:
@markmandel I've said that scale property is not useful as we can just use a nullable field, @aLekSer said this was from the design ticket, WDYT ?

Reply:
If we go this way, then we have to have:

```suggestion
 	Replicas *int32 `json:"replicas,omitempty"`
```

Otherwise, this will always get populated with a `0`.

That being said - thinking through usage scenarios:

```go
review.Response.Scale = false
```

vs

```go
review.Response.Replicas = nil
```

To disable scaling on the return value. Just thinking about what makes more sense when reading the code.

I was basing my design on https://godoc.org/k8s.io/api/admission/v1beta1#AdmissionResponse which uses `Allowed` as a Boolean.

I'm more inclined towards the boolean, since it's more explicit. What do you all think?

Reply:
Regarding Mark comment, we can follow AdmissionResponse approach which is more explicit, since the size of json is not crucial in this case.

Reply:
Alright sorry for the confusion !